vllm/model_executor/models/qwen2_moe.py

```diff
@@ -407,13 +408,31 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
         ]
 
         expert_params_mapping = [
+            # These are the weight scales for the experts
+            # (param_name, weight_name, expert_id, shard_id)
+            ("experts.w13_scale" if weight_name in ["gate_proj", "up_proj"
+                                                    ] else "experts.w2_scale",
+             f"experts.{expert_id}.{weight_name}.weight_scale", expert_id,
+             shard_id) for expert_id in range(self.config.num_experts)
+            for shard_id, weight_name in enumerate(
+                ["gate_proj", "down_proj", "up_proj"])
+        ] + [
```

Comment:
For future thought/refactor: It would be nice to specify which strings are needed for FusedMoE and which are needed for the model checkpoint. This would make it more explicit how the remapping happens. Maybe making strings like `"experts.w13_scale"` and "experts.w2_scale" enums defined in FusedMoE could be a start?

Reply:
I think this is a good idea.